pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -333,23 +333,20 @@ func GetHugepageLimitsFromResources(resources v1.ResourceRequirements) []*runtim
 	}
 
 	requiredHugepageLimits := map[string]uint64{}
-	for resourceObj, amountObj := range resources.Limits {
-		if !v1helper.IsHugePageResourceName(resourceObj) {
-			continue
-		}
 
-		pageSize, err := v1helper.HugePageSizeFromResourceName(resourceObj)
-		if err != nil {
-			klog.InfoS("Failed to get hugepage size from resource", "object", resourceObj, "err", err)
-			continue
+	// When hugepage limits are specified at pod level, the container limit will
```

Comment:
The comment seems confusing. If container-level values are set, they are used as-is. Use of "precedence" in this case can be confusing. 

Nit and for clarity: Is it possible to populate requiredHugepageLimits using container-level values first, and then propagate pod-level limits to container-level limits only if the the respective key i.e. sizeString doesn't exist in requiredHugepageLimits.

Reply:
Refer to answer in https://github.com/kubernetes/kubernetes/pull/131089#discussion_r2173549963

Updated comment to:
```
	// When hugepage limits are specified at pod level and no hugepage limits are
	// specified at container level, the container's cgroup will reflect the pod level limit.
```